pydantic_ai_slim/pydantic_ai/agent.py

```diff
@@ -854,9 +859,13 @@ async def _handle_streamed_model_response(
                 else:
                     messages.append(self._unknown_tool(call.tool_name))
 
-            with _logfire.span('running {tools=}', tools=[t.get_name() for t in tasks]):
-                task_results: Sequence[_messages.Message] = await asyncio.gather(*tasks)
-                messages.extend(task_results)
+            with _logfire.span('running {tools=}', tools=[t.get_name() for t in tasks]) as span:
+                try:
+                    task_results: Sequence[_messages.Message] = await asyncio.gather(*tasks)
+                    messages.extend(task_results)
+                except _exceptions.StopAgentRun as e:
+                    span.set_attribute('stop_agent_run', e.tool_name)
+                    return _MarkFinalResult(data=e.result), []
```

Comment:
This is not correct, and basically should be a type error, but the problem is that `e.result` has type `Any`. However, I am not sure what's going on here so not sure how to tweak things to fix this. Maybe we can discuss tomorrow.